### DISCUSSION TITLE
Can use "Query" and "Path" fields in a Pydantic model but not "Header"

### DISCUSSION THREAD

**Discussion by dmcneil** [fastapi/fastapi#9514] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import pydantic
import uvicorn
from fastapi import FastAPI, Depends, Header, Path, Query


# Just a regular class with the fields being declared in the constructor.
class NormalExample:
    def __init__(
        self,
        id: str = Path(),
        foo: str = Query(),
        bar: str = Header(),
    ):
        self.id = id
        self.foo = foo
        self.bar = bar


# Pydantic model version.
class ModelExample(pydantic.BaseModel):
    id: str = Path()
    foo: str = Query()
    bar: str = Header()


app = FastAPI()


@app.get("/normal/{id}")
async def normal(example: NormalExample = Depends()):
    print("id =", example.id)
    print("foo = ", example.foo)
    print("bar = ", example.bar)


@app.get("/model/{id}")
async def model(example: ModelExample = Depends()):
    print("id =", example.id)
    print("foo = ", example.foo)
    print("bar = ", example.bar)


uvicorn.run(app=app, host="0.0.0.0", port=8080)
```

### Description

Is it possible to use the `Header` type as field in a class that extends from `pydantic.BaseModel`? It looks like you can use both `Path` and `Query` fields without any issues but `Header` gets treated as a query parameter.

Referring to the code example, if you would hit the first route via:

```shell
curl -H 'bar: BAR' 'http://localhost:8080/normal/12345?foo=FOO'
```

Then the following prints:

```
id = 12345
foo =  FOO
bar =  BAR
```

If you omit the `bar` header you get the following error response which is expected because it's required:

```json
{
    "detail": [
        {
            "loc": [
                "header",
                "bar"
            ],
            "msg": "field required",
            "type": "value_error.missing"
        }
    ]
}
```

Note the `detail.loc` field contains `header`.

If you request against the second route via:

```shell
curl -H 'bar: BAR' 'http://localhost:8080/model/12345?foo=FOO'
```

The response is:

```json
{
    "detail": [
        {
            "loc": [
                "query",
                "bar"
            ],
            "msg": "field required",
            "type": "value_error.missing"
        }
    ]
}
```

Where it appears the `ModelExample.bar` is being treated as a required query parameter. To confirm, you can do:

```shell
curl 'http://localhost:8080/model/12345?foo=FOO&bar=BAR'
```

Which results in no error and the following being printed:

```
id = 12345
foo =  FOO
bar =  BAR
```

### Operating System

macOS

### FastAPI Version

0.95.1

### Python Version

3.10.2

**Comment by djimontyp** [comment]
@dmcneil Try this: 
```
class ModelExample(pydantic.BaseModel):
    id: Annotated[str, Path()]
    foo: Annotated[str, Query()]
    bar: Annotated[str, Header()]
```

**Reply by DeoLeung** [reply]
https://stackoverflow.com/questions/75998227/how-to-define-query-parameters-using-pydantic-model-in-fastapi

wrap into `Field` helps

**Reply by tatianak22608** [reply]
> https://stackoverflow.com/questions/75998227/how-to-define-query-parameters-using-pydantic-model-in-fastapi
> 
> 
> 
> wrap into `Field` helps
